src/plugins/intel_cpu/src/emitters/snippets/x64/jit_fill_emitter.cpp

```diff
@@ -59,10 +59,16 @@ void jit_fill_emitter::emit_isa(const std::vector<size_t> &in, const std::vector
     Vmm src_vmm = Vmm(in[0]);
     Vmm dst_vmm = Vmm(out[0]);
 
-    if (is_full_reg())
+    const size_t supported_et_size = 4;
+    const auto register_capacity = (src_vmm.getBit() / 8) / supported_et_size;
+    if (offset == register_capacity) {
+        if (src_vmm.getIdx() != dst_vmm.getIdx())
+            h->uni_vmovups(dst_vmm, src_vmm);
```

Comment:
Earlier, we inserted Fill only in tail loop body. However, to do that, we had to specify a [special rt_info to the ngraph node inputs with all necessary information](https://github.com/openvinotoolkit/openvino/pull/21303/files#diff-e0eed067563b2ee171c348c6aed00efd7dbfc7a03bf5247a35fec08591e3d193L116-L125).
We decided to remove this workaround. To do that, we decided to insert Fill in the main body, and change offset via loop handler in tail body. Since full register is used in the main body, we insert Fill with `offset=vector_size` in `SoftmaxDecomposition`. At first, we wanted to remove Fill from the main body via another handler, however, we met with `AssignRegister` limitations: it doesn't support inplace logic. So we can't just remove Fill from the main body. As a temporal solution, we decided to optimize `offset=vector_size` case on the emitter level (by adding `vmovups` instead of filling logic). When inplace is supported in `AssignRegisters` pass (ticket 126270), we will be able to remove Fill from the main body via loop handler.

Reply:
The comment is added, the ticket was also updated. Thanks!